### ISSUE TITLE
Hyperparameter tuning

### ISSUE THREAD

**Issue by inoue0426** [cannin/gsoc_2023_pytorch_pathway_commons#22]
Hyperparameters
- number of hidden layers. You can set different # of layers for each hidden layer.
- conv layer(GCN, GAT, GraphSAGE etc.) [layer](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#convolutional-layers)
- dropout ratio
- activation function (ReLU, LeakyReLU, Tanh etc.) [functions](https://pytorch.org/docs/stable/nn.html#non-linear-activations-weighted-sum-nonlinearity)
- Add normalization layer (GraphNorm, BatchNorm etc) [GraphNorm](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.norm.GraphNorm.html)
- Dropout for all layers
- mean_pool (global, average, etc.) [Pooling layers](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#pooling-layers)
- learning rate
- optimizer (Adam, RMSprop, SGC etc) [optimizers](https://pytorch.org/docs/stable/optim.html#module-torch.optim)

You can use the optuna. [optuna](https://optuna.readthedocs.io/en/stable/index.html)
